qemu/tests/ioeventfd.py

```diff
@@ -41,13 +42,36 @@ def run(test, params, env):
             2.6 Check the ioeventfd=on via /proc/$PID/fd/.
             2.7 Compare the output of 'ls -l /proc/$PID/fd/', the fds with
                 "off" should be less than the one with "on".
+        Scenario 3:
+            3.1 Boot guest with ioeventfd=off and port attched to virtio_serial_pci.
+            3.2 Execute info qtree in QMP monitor, info qtree should show the
+                ioeventfd = false.
+            3.3 Check the ioeventfd=off via /proc/$PID/fd/.
+            3.4 Transfer data via the virtserialport.
+            3.5 Boot guest with ioeventfd=on attched to one virtio_serial_pci.
+            3.6 Execute info qtree in QMP monitor, info qtree should show the
+                ioeventfd = true.
+            3.7 Check the ioeventfd=on via /proc/$PID/fd/.
+            3.8 Compare the output of 'ls -l /proc/$PID/fd/', the fds with
+                "off" should be less than the one with "on".
+            3.9 Transfer data via the virtserialport.
 
     :param test: QEMU test object
     :param params: Dictionary with the test parameters
     :param env: Dictionary with test environment.
     """
-    def _set_ioeventfd_options():
-        """ Set the ioeventfd options. """
+    def _set_ioeventfd_options(ioeventfd):
+        """
+        Set the ioeventfd options.
+
+        :param ioeventfd: the value of parameter ioeventfd
+        :return: device id with parameter ioeventfd
+        """
+        type_dev = params.get("type_dev")
+        if type_dev == "virtio_serial":
+            params['virtio_serial_extra_params'] = ioeventfd
+            dev_id = 'virtio_serial_pci0'
+            return dev_id
         if params['drive_format'] == 'virtio':
```

Comment:
We do not test both serial and block device ioenventfd at the same time.
use `else` please

Reply:
The operation of `return` will separate the two tests.

Reply:
yes, you are right,  i missed the `return`.   but  it is dupl `return`, no need to call it for every scenario.

Reply:
Thanks and update.